app/square_size.go

```diff
@@ -19,6 +19,6 @@ func (app *App) GovSquareSizeUpperBound(ctx sdk.Context) int {
 	}
 
 	gmax := int(app.BlobKeeper.GovMaxSquareSize(ctx))
-	hardMax := appconsts.SquareSizeUpperBound(app.AppVersion(ctx))
+	hardMax := appconsts.SquareSizeUpperBound(ctx.BlockHeader().Version.App)
```

Comment:
Actually I don't get how this fixes it? The above fix using appVersion should work.

AppVersion calls the consensus parameters which includes the correct app version (the one we should rely on) because we don't always have a header in the context to compare with